internal/ark/events/cluster_events.go

```diff
@@ -0,0 +1,49 @@
+// Copyright © 2018 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package events
+
+type clusterEvents interface {
+	NotifyClusterDeleted(fn interface{})
+}
+
+type eventBus interface {
+	SubscribeAsync(topic string, fn interface{}, transactional bool) error
+}
+
+type clusterEventBus struct {
+	eb eventBus
+}
+
+const (
+	clusterCreatedTopic = "cluster_created"
```

Comment:
Instead of introducing constants here for `cluster_created` and `cluster_deleted` can't we just create some global costs for these and use everywhere? There are constants for these event literals in `internal/monitor/events.go`, `cluster/events.go`. (see `OrganizationRegisteredTopic`)

Reply:
I had a conversation about this with @sagikazarmark yesterday. The main reason is that these are separate consumers and as such is better to have separate interfaces for each, this gives us the ability to move those consumers out from pipeline easier. Surely Mark can explain it more clearly.

Reply:
Please remove the word "global" from your dictionaries. 🙂